packages/salesforce-adapter/src/filters/custom_objects_instances.ts

```diff
@@ -0,0 +1,134 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { InstanceElement, ObjectType, Element, isObjectType, Field, ElemID, Values } from '@salto-io/adapter-api'
+import { SalesforceClient } from 'index'
+import { SALESFORCE, RECORDS_PATH, INSTALLED_PACKAGES_PATH } from '../constants'
+import { FilterCreator } from '../filter'
+import { apiName, isCustomObject, Types } from '../transformers/transformer'
+import { getNamespace } from './utils'
+
+const buildQueryString = (type: ObjectType): string => {
+  const selectStr = Object.values(type.fields)
+    .map((field: Field) => {
+      const fieldType = field.type
+      if (isObjectType(fieldType)
+        && fieldType.elemID.isEqual(Types.compoundDataTypes.Name.elemID)) {
+        return Object.keys(fieldType.fields).join(',')
+      }
+      return apiName(field, true)
+    }).join(',')
+  return `SELECT ${selectStr} FROM ${apiName(type)}`
+}
+
+const consumeAsyncIterable = async <Values, InstanceElement>(
+  itr: { [Symbol.asyncIterator]: () => AsyncIterator<Values[]> },
+  transformer: (t: Values[]) => InstanceElement[],
+): Promise<Array<InstanceElement>> => {
+  const res: InstanceElement[] = []
+  const iter = itr[Symbol.asyncIterator]()
+  const next = async (): Promise<void> => {
+    const curr = await iter.next()
+    if (curr.done) return undefined
+    if (_.isArray(curr.value) && curr.value.length > 0) {
+      res.push(...transformer(curr.value))
+    }
+    return next()
+  }
+  await next()
+  return res
+}
+
+const getObjectInstances = async (
+  client: SalesforceClient,
+  object: ObjectType
+): Promise<Array<InstanceElement>> => {
+  const recordsToInstances = (records: Values[]): InstanceElement[] => {
+    const recordToInstance = (record: Values): InstanceElement => {
+      const getInstancePath = (instanceName: string): string[] => {
+        const objectNamespace = getNamespace(object)
+        if (objectNamespace) {
+          return [SALESFORCE, INSTALLED_PACKAGES_PATH, objectNamespace,
+            RECORDS_PATH, object.elemID.typeName, instanceName]
+        }
+        return [SALESFORCE, RECORDS_PATH, object.elemID.typeName, instanceName]
+      }
+
+      return new InstanceElement(
+        // TODO: Handle elemID with additional logic
+        new ElemID(SALESFORCE, record.Id).name,
```

Comment:
why is the `ElemID` needed here? isn't this the same as just using `record.Id`?
```suggestion
        record.Id,
```